File: GreenopsBackend/Models/Complaint.cs
using System.ComponentModel.DataAnnotations;

namespace Greenops.Models
{
    public class Complaint
    {
        [Key]
        public int ComplaintId { get; set; }

        public int UserId { get; set; }

        public string ComplaintDescription { get; set; }

        public string ProofImage { get; set; }

        public DateTime ComplaintDate { get; set; }

        public string? Status { get; set; }
    }
}

File: GreenopsBackend_JavaSpring/src/main/java/com/greenops/entity/Complaint.java
package com.greenops.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "Complaints")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Complaint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ComplaintId")
    private Integer complaintId;

    @Column(name = "UserId", nullable = false)
    private Integer userId;

    @Column(name = "ComplaintDescription", nullable = false)
    private String complaintDescription;

    @Column(name = "ProofImage")
    private String proofImage;

    @Column(name = "ComplaintDate", nullable = false)
    private LocalDateTime complaintDate;

    @Column(name = "Status")
    private String status;
}